integration-tests/src/test/scala/VideoUploadTests.scala

```diff
@@ -0,0 +1,153 @@
+import java.io.{FilterInputStream, InputStream}
+import java.time.Instant
+import java.util.UUID
+
+import com.amazonaws.auth.{AWSStaticCredentialsProvider, BasicSessionCredentials}
+import com.amazonaws.regions.{Region, Regions}
+import com.amazonaws.services.s3.AmazonS3Client
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.google.common.io.ByteStreams
+import com.gu.media.util.TestFilters
+import com.squareup.okhttp.Response
+import integration.IntegrationTestBase
+import integration.services.Config
+import org.scalatest.CancelAfterFailure
+import org.scalatest.time.{Minutes, Seconds, Span}
+import play.api.libs.json.{JsArray, JsValue, Json}
+import services.AWS
+
+class VideoUploadTests extends IntegrationTestBase with CancelAfterFailure {
+  val s3 = accountCredentialsS3Client()
+
+  var atomId: String = ""
+  var apiEndpoint: String = ""
+
+  var uploadId: String = ""
+  var uploadBucket: String = ""
+  var uploadParts = List.empty[(String, Long, Long)]
+  var uploadUri: Option[String] = None
+  var completeKey: String = ""
+
+  override def beforeAll(): Unit = {
+    val aliveResponse = gutoolsGet(targetBaseUrl)
+    aliveResponse.code() should be (200)
+    aliveResponse.body().string() should include ("video")
+
+    val json = generateJson(
+      title = s"${TestFilters.testAtomBaseName}-${UUID.randomUUID().toString}",
+      description = "test atom",
+      category = "News",
+      channelId = Config.channelId,
+      youtubeCategoryId = Config.youtubeCategoryId,
+      expiryDate = Instant.now.toEpochMilli + (100 * 60 * 60 * 24)
+    )
+
+    val response = gutoolsPost(s"$targetBaseUrl/api2/atoms", jsonBody(json))
+
+    response.code() should be(201)
+
+    atomId = (Json.parse(response.body().string()) \ "id").get.as[String]
+
+    addAtomToStore(atomId)
+
+    apiEndpoint = apiUri(atomId)
+
+    eventually {
+      gutoolsGet(apiEndpoint).code() should be(200)
+    }
```

Comment:
This completely duplicates the steps of the atom creation tests. Might be good to pull this out to a separate function, probably in IntegrationTestBase, for easier maintenance